src/chat.rs

```diff
@@ -252,7 +252,7 @@ impl ChatId {
         let chat_id = match ChatIdBlocked::lookup_by_contact(context, contact_id).await? {
             Some(chat) => {
                 if create_blocked == Blocked::Not && chat.blocked != Blocked::Not {
-                    chat.id.unblock_ex(context, Nosync).await?;
+                    chat.id.set_blocked(context, Blocked::Not).await?;
```

Comment:
This doesn't change anything, but only to avoid Rust error about recursive calls of async functions

Reply:
it's not so easy to guess that set_blocked() and unblock_ex both do "Nosync" but seems to be the case, indeed.

Reply:
There's a rule now: all private and `pub(crate)` functions don't do any sync unless `Sync` is explicitly passed. But only public ones because they are called on user actions. I plan to keep this rule further